### PR TITLE
0669 update config.js schema

### PR DESCRIPTION
**Pull request by tforster**
The most obvious and compelling change introduced in this PR is the addition of S3+CloudFront as a deployable target. Note that Amplify has been (temporarily?) removed because of instabilities in the service.

In addition to S3+CloudFront the PR implements improved configuration management via the new webproducer.yml file and automatic environment variable expansion. The YAML format is easier to read than the previous config.js and unlike JSON, supports inline comments. Webproducer.yml can also accept environment variables in the format `key: ${env:variable-name}`.

WebProducer performance has been boosted in that the previous brute force approach to **copy-all-built-files** has been replaced with a more efficient copy of **changed-files-only**.

Less visible but equally important in this PR is the multitude of under-the-hood changes, tweaks and improvements. These changes are listed in the messages for the accompanying commits.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -2,7 +2,6 @@
   "extends": [
     "prettier"
   ],
-  "parser": "babel-eslint",
```
Comment (tforster) [inline .eslintrc.json:5]: This is a redundant requirement given the capabilities of eslint. It still exists in some of our other repos and should be removed if/when we find it. The accompanying babel-eslint devDependency package should also be removed from package.json.

File: package.json [hunk 2]
```diff
@@ -51,8 +51,10 @@
     "handlebars": "^4.7.6",
     "jszip": "^3.2.2",
     "micromatch": "^4.0.2",
+    "mime": "^2.4.6",
```
Comment (tforster) [inline package.json:54]: Zero deps

File: .eslintrc.json (hunk 1 above)
Comment (webdivelement) [reply .eslintrc.json:5]: Will do, thanks!

File: src/StreamUtils.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  /**
+   * Helper function to set various additional properties on a VinylFile
+   *
+   * @static
+   * @param {*} vinylFile
+   * @param {*} options
+   * @returns
+   * @memberof Metafy
```
Comment (webdivelement) [inline src/StreamUtils.js:47]: Which class is Metafy?  I see it referenced in the comments a few times.

Comment (tforster) [reply src/StreamUtils.js:47]: Hmm mobile interface did not give me this option to reply (now on desktop).

Metafy was a poor name that I used as a placeholder with the intent of updating. Which I did, to StreamUtils, which is only marginally better.

I have been sketching out a revamped layout and naming convention that I will ticket into Boards soon.

**Pull request merged**